src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,30 +1,68 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
-    </p>
-  </form>
-);
+import { useCallback, useEffect, useState } from 'react';
+import debounce from 'lodash.debounce';
+
+type Props = {
+  setFilter: (filter: string) => void;
+  setSearch: (search: string) => void;
+};
+
+export const TodoFilter: React.FC<Props> = ({ setFilter, setSearch }) => {
+  const [currentSearch, setCurrentSearch] = useState('');
+
+  const debouncedSetSearch = useCallback(debounce(setSearch, 500), []);
+
+  useEffect(() => {
+    debouncedSetSearch(currentSearch.trim());
+
+    return () => {
+      debouncedSetSearch.cancel();
+    };
+  }, [currentSearch, debouncedSetSearch]);
+
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            data-cy="statusSelect"
+            onChange={e => setFilter(e.target.value)}
```

Comment:
It's recommended to follow naming conventions for event handler functions. Consider renaming the inline function to something like `handleFilterChange`.